paasta_tools/cli/cmds/logs.py

```diff
@@ -89,18 +89,29 @@ def add_subparser(subparsers) -> None:
         "--service",
         help="The name of the service you wish to inspect. Defaults to autodetect.",
     ).completer = lazy_choices_completer(list_services)
-    components_help = "A comma separated list of the components you want logs for."
     status_parser.add_argument(
-        "-C", "--components", help=components_help
-    ).completer = lazy_choices_completer(LOG_COMPONENTS.keys)
-    cluster_help = "The clusters to see relevant logs for. Defaults to all clusters to which this service is deployed."
-    status_parser.add_argument(
-        "-c", "--clusters", help=cluster_help
+        "-c",
+        "--clusters",
+        help="The clusters to see relevant logs for. Defaults to all clusters to which this service is deployed.",
     ).completer = completer_clusters
-    instance_help = "The instances to see relevant logs for. Defaults to all instances for this service."
     status_parser.add_argument(
-        "-i", "--instances", help=instance_help
+        "-i",
+        "--instances",
+        help="The instances to see relevant logs for. Defaults to all instances for this service.",
     ).completer = completer_clusters
+    status_parser.add_argument(
+        "-C",
+        "--components",
+        type=lambda s: set(s.split(",")),
+        default=set(DEFAULT_COMPONENTS),
+        help=(
+            "A comma-separated list of the components you want logs for. "
+            "PaaSTA consists of 'components' such as builds and deployments, "
+            "for each of which we collect logs for per service. "
+            "See below for a list of components. "
+            f"Defaults to {DEFAULT_COMPONENTS}."
```

Comment:
could also do `"Defaults to %(default)s"` to have argparse handle interpolating for you if you wanted to :)

Reply:
TIL